### ISSUE TITLE
ArgsTable of={Component} cannot find propTypes

### ISSUE THREAD

**Issue by jasongaare** [storybookjs/storybook#18137]
**Describe the bug**

I've followed directions outlined here https://github.com/storybookjs/storybook/issues/12576 to export my wrapped components directly, however my `<ArgsTable of={Component/}>` component still renders that "No inputs found for this component"

However, when I console.log the component itself, the propTypes + defaultProps seem to be plainly available. 

The way I read this, it should "just work"?
> If you need, you can customize what is displayed in the ArgsTable by extending the ArgTypes data, unless you're using the ArgsTable of={component} />. In this case, Storybook will infer the data automatically from the component.

**Screenshot**

![Screen Shot 2022-05-04 at 1 49 49 PM](https://user-images.githubusercontent.com/15896334/166805210-f5eb4d46-227c-436b-a729-27302727fbc7.png)

I've scoured the issues here, tried various webpack configs, babel configs but nothing seems to clear my issue. I'm working in a monorepo with Expo + React Native + React + storybook, so there's likely some wires crossed somewhere. 

**System**
Please paste the results of `npx sb@next info` here.

<details>
  <summary>Click for System Info</summary>

**Environment Info:**

  System:
    OS: macOS 12.3.1
    CPU: (10) x64 Apple M1 Pro
  Binaries:
    Node: 12.22.0 - ~/.nvm/versions/node/v12.22.0/bin/node
    Yarn: 1.22.18 - /opt/homebrew/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v12.22.0/bin/npm
  Browsers:
    Chrome: 100.0.4896.127
    Firefox: 100.0
    Safari: 15.4
</details>

**Comment by dmackerman** [comment]
Yeah, I can't get this to work either. Not doing anything fancy. :\

**Comment by shenyi-dra** [comment]
+1

**Comment by samuelhorn** [comment]
Same thing here!

**Comment by RwwL** [comment]
We've got a scenario right now where this is happening for a single new component in a storybook with dozens of other components that we've been building for years, *and* it's only happening for about half of the people who've checked out the repo.

**Comment by shilman** [comment]
Do you a have a reproduction repo you can share? If not, can you create one? See [how to create a repro](https://storybook.js.org/docs/react/contribute/how-to-reproduce). Thank you! 🙏

**Comment by sag1v** [comment]
Having the same issue during a migration to v7 from v6. 
Can't share the repo unfortunately as it is private.

**Comment by RwwL** [comment]
@sag1v In the component whose `ArgsTable` doesn't work in a story, are you `export`ing anything else at all? That wound up being the solution for us: as soon as we moved that utility into its own file and imported it into the component instead, the `ArgsTable` started working for everyone on the team.

**Comment by sag1v** [comment]
@jasongaare Did you ever solve this issue? i'm still struggling to find a solution (or to reproduce with a clean sandbox)

**Comment by jasongaare** [comment]
We never did find a solution. We currently manually define all our prop types and feed them to an ArgsTable that way

**Comment by sag1v** [comment]
> We never did find a solution. We currently manually define all our prop types and feed them to an ArgsTable that way

@jasongaare What version of react do you have?

**Comment by sag1v** [comment]
I think I've managed to reproduce this issue (finally!!!).

It turns out, that whenever there is a babel config e.g: `.babelrc.json` with the `'@babel/preset-env'` preset set, it breaks the ArgTypes/argTable component.

See [this sandbox](https://stackblitz.com/edit/github-eeqtnm).

When removing the preset it works fine.

CC @shilman

**Comment by sag1v** [comment]
> I think I've managed to reproduce this issue (finally!!!).
> 
> It turns out, that whenever there is a babel config e.g: `.babelrc.json` with the `'@babel/preset-env'` preset set, it breaks the ArgTypes/argTable component.
> 
> See [this sandbox](https://stackblitz.com/edit/github-eeqtnm).
> 
> When removing the preset it works fine.
> 
> CC @shilman

**UPDATE** 
It seems that the issue is related to the `targets` property and also specifically to the `chrom: 100` settings 🤔 
Any other settings seems to work.
See this video:

https://github.com/storybookjs/storybook/assets/9153149/ff03e740-cf61-4414-806a-6fbb9bee0112

 Not sure what is this black voodoo all about 🤷
